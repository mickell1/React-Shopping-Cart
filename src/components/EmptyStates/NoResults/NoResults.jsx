import React from "react";
import noresults from '../../../images/no-results.png';

import './NoResults.scss';

const NoResults = () => {
  return (
    <div className="products">
      <div className="no-results">
        <img
          src={noresults}
          alt="No results"
        />
        <h2>Sorry, no products matched your search!</h2>
        <p>Enter a different keyword and try.</p>
      </div>
    </div>
  );
};

export default NoResults;
