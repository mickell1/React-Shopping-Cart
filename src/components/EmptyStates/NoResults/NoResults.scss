.products-wrapper {
  padding-top: 150px;
  animation: slideUp 300ms linear;
  animation-delay: 150ms;
  @media (max-width: 991px) {
    padding-top: 68px;
  }
  @media (max-width: 480px) {
    padding-top: 52px;
  }
}
.products {
  display: flex;
  display: -webkit-flex;
  flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  padding: 16px;
  max-width: 960px;
  margin: 0 auto;
  @media (max-width: 768px) {
    padding: 8px;
  }
  .product {
    min-height: 355px;
    flex-basis: calc(25% - 32px);
    -webkit-flex-basis: calc(25% - 32px);
    @media (max-width: 768px) {
      flex-basis: calc(33% - 16px);
      -webkit-flex-basis: calc(33% - 16px);
      margin: 8px;
    }
    @media (max-width: 480px) {
      flex-basis: calc(50% - 16px);
      -webkit-flex-basis: calc(50% - 16px);
      min-height: 313px;
      .product-name {
        height: 20px;
        overflow: hidden;
      }
      .product-action {
        padding: 16px 8px;
      }
    }
    @media (max-width: 320px) {
      flex-basis: 100%;
      -webkit-flex-basis: 100%;
    }
  }
}

// Empty states
.no-results {
  text-align: center;
  max-width: 520px;
  margin: 0 auto;
  img {
    width: 320px;
    margin: 16px 0 32px 0;
    opacity: 0.33;
  }
  h2 {
    margin-bottom: 16px;
    color: #666;
  }
  p {
    color: #999;
  }
}