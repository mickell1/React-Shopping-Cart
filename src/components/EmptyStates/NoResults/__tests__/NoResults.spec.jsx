import React from 'react';
import { shallow, mount } from 'enzyme';
import NoResults from '../NoResults';

describe('NoResults', () => {

  let wrapper;

  beforeEach(()=>{
    wrapper = shallow(<NoResults /> )  
  })

  it('should render the NoResults Component correctly', () => {   
    expect(wrapper).toMatchSnapshot();
  });

  it('should render classname', () => {
    wrapper = mount(
     <NoResults />
   );

   expect(wrapper.find('div').first().hasClass('products')).toEqual(true);
 });

  it('should render h2', () => {
    wrapper = mount(<NoResults />)
    expect(wrapper.find('h2').text()).toBe('Sorry, no products matched your search!')
  })

  it('should render p', () => {
    wrapper = mount(<NoResults />)
    expect(wrapper.find('p').text()).toBe('Enter a different keyword and try.')
  })
});