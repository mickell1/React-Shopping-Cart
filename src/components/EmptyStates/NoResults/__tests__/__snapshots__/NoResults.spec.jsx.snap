// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoResults should render the NoResults Component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoResults />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="no-results"
      >
        <img
          alt="No results"
          src="no-results.png"
        />
        <h2>
          Sorry, no products matched your search!
        </h2>
        <p>
          Enter a different keyword and try.
        </p>
      </div>,
      "className": "products",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="No results"
            src="no-results.png"
          />,
          <h2>
            Sorry, no products matched your search!
          </h2>,
          <p>
            Enter a different keyword and try.
          </p>,
        ],
        "className": "no-results",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "No results",
            "src": "no-results.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sorry, no products matched your search!",
          },
          "ref": null,
          "rendered": "Sorry, no products matched your search!",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enter a different keyword and try.",
          },
          "ref": null,
          "rendered": "Enter a different keyword and try.",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="no-results"
        >
          <img
            alt="No results"
            src="no-results.png"
          />
          <h2>
            Sorry, no products matched your search!
          </h2>
          <p>
            Enter a different keyword and try.
          </p>
        </div>,
        "className": "products",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="No results"
              src="no-results.png"
            />,
            <h2>
              Sorry, no products matched your search!
            </h2>,
            <p>
              Enter a different keyword and try.
            </p>,
          ],
          "className": "no-results",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "No results",
              "src": "no-results.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sorry, no products matched your search!",
            },
            "ref": null,
            "rendered": "Sorry, no products matched your search!",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Enter a different keyword and try.",
            },
            "ref": null,
            "rendered": "Enter a different keyword and try.",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
